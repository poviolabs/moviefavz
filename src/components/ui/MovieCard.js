import React from 'react';
import PropTypes from 'prop-types';

import styled from 'styled-components';

import { Card } from 'antd';

const { Meta } = Card;

const StyledCard = styled(Card)`
  width: 100%;

  .ant-card-cover {
    height: 260px;

    img {
      object-fit: cover;
      width: 100%;
      height: 100%;
    }
  }
`;

const MovieCard = ({ Title, Poster, Type, Year, imdbID, onPress }) => {
  return (
    <StyledCard
      hoverable
      cover={<img alt={Title} src={Poster} />}
      onClick={() => (onPress ? onPress(imdbID) : null)}
    >
      <Meta title={Title} description={`${Year} â€¢ ${Type}`} />
    </StyledCard>
  );
};

MovieCard.propTypes = {
  Title: PropTypes.string,
  Poster: PropTypes.string,
  Type: PropTypes.string,
  Year: PropTypes.string,
  imdbID: PropTypes.string,
  onPress: PropTypes.func,
};

export default MovieCard;
