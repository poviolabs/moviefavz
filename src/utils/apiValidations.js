const validResponse = (status) => status && status >= 200 && status < 300;

export default { validResponse };
