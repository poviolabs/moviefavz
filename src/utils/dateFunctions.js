import dayjs from 'dayjs';

const getUnixDay = (timestamp = Date.now()) => {
  return dayjs(timestamp).startOf('date').valueOf();
};

const fillDaysToMatch = (num, data) => {
  if (!data.length) {
    return [];
  }
  const [lastDate] = data.reverse();

  return Array.from('x'.repeat(num - data.length))
    .map((_, i) => i + 1)
    .reverse()
    .map((i) => {
      return {
        value: 0,
        timestamp: dayjs(lastDate.timestamp).subtract(i, 'day').valueOf(),
      };
    });
};

export default { getUnixDay, fillDaysToMatch };
