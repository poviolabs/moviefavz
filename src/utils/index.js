export { default as arrayFunction } from './arrayFunctions';
export { default as apiValidations } from './apiValidations';
