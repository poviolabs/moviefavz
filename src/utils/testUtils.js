import React from 'react';
import { render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';

import { ThemeProvider } from 'styled-components';
import { theme } from '../styles';

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <BrowserRouter>{children}</BrowserRouter>
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
