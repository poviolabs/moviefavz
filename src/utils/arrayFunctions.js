import dateFunctions from './dateFunctions';

const groupArr = (data, size) => {
  const group = [];
  for (let i = 0, j = 0; i < data.length; i++) {
    if (i >= size && i % size === 0) {
      j++;
    }
    group[j] = group[j] || [];
    group[j].push(data[i]);
  }
  return group;
};

const groupByDay = (data, key = 'timestamp') => {
  const groupedByKey = data.reduce((obj, item) => {
    const timestamp = item[key];
    const day = dateFunctions.getUnixDay(timestamp);

    if (Object.prototype.hasOwnProperty.call(obj, day)) {
      obj[day].value += 1;
    } else {
      obj[day] = {
        value: 1,
        timestamp,
      };
    }
    return obj;
  }, {});
  return Object.values(groupedByKey);
};

export default { groupArr, groupByDay };
