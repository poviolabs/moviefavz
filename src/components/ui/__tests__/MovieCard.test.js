import React from 'react';
import { render } from '../../../utils/testUtils';

import MovieCard from '../MovieCard';

describe('<MovieCard />', () => {
  it('Renders the Component', () => {
    const props = {
      Title: 'Card Title',
      Poster: 'N/A',
      Type: 'movie',
      Year: '2020',
      imdbID: 'tt7653',
      onPress: jest.fn(),
      favorited: true,
      onFavoritePress: jest.fn(),
    };
    const container = render(<MovieCard {...props} />);
    expect(container).toMatchSnapshot();
  });
});
