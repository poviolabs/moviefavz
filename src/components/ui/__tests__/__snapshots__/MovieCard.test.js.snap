// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCard /> Renders the Component 1`] = `undefined`;
